lib/mayaUsd/ufe/UsdTransform3dMayaXformStack.cpp

```diff
@@ -433,111 +433,97 @@ UsdTransform3dMayaXformStack::translateCmd(double x, double y, double z)
-              TF_AXIOM(usdSceneItem);
-              auto attr = usdSceneItem->prim().GetAttribute(attrName);
-
-              // return an invalid XformOp if the attribute edit is not allowed.
-              std::string errMsg;
-              if (!MayaUsd::ufe::isAttributeEditAllowed(attr, &errMsg)) {
-                  MGlobal::displayError(errMsg.c_str());
-                  return UsdGeomXformOp();
-              }
-
-              return UsdGeomXformOp(attr);
-          })
-        : OpFunc([opSuffix = getTRSOpSuffix(), setXformOpOrderFn = getXformOpOrderFn(), v](
-                     const BaseUndoableCommand& cmd) {
-              // Use notification guard, otherwise will generate one notification
-              // for the xform op add, and another for the reorder.
-              InTransform3dChange guard(cmd.path());
-              auto usdSceneItem = std::dynamic_pointer_cast<UsdSceneItem>(cmd.sceneItem());
-              TF_AXIOM(usdSceneItem);
-              UsdGeomXformable xformable(usdSceneItem->prim());
-
-              // At this point we already know that writing to attribute is going
-              // to succeed but we do not know if writing to "transform op order" succeed since
-              // we could be a weaker layer.
-              std::string errMsg;
-              if (!MayaUsd::ufe::isAttributeEditAllowed(xformable.GetXformOpOrderAttr(), &errMsg)) {
-                  MGlobal::displayError(errMsg.c_str());
-                  return UsdGeomXformOp();
-              }
-
-              auto r = xformable.AddRotateXYZOp(UsdGeomXformOp::PrecisionFloat, opSuffix);
-              TF_AXIOM(r);
-              r.Set(v);
-              auto result = setXformOpOrderFn(xformable);
-              TF_AXIOM(result);
-
-              return r;
-          });
+
+    auto f = OpFunc(
+        [attrName, opSuffix = getTRSOpSuffix(), setXformOpOrderFn = getXformOpOrderFn(), v](
+            const BaseUndoableCommand& cmd) {
+            auto usdSceneItem = std::dynamic_pointer_cast<UsdSceneItem>(cmd.sceneItem());
+            TF_AXIOM(usdSceneItem);
+            auto attr = usdSceneItem->prim().GetAttribute(attrName);
+            if (attr) {
+                return UsdGeomXformOp(attr);
+            } else {
+                // Use notification guard, otherwise will generate one notification
+                // for the xform op add, and another for the reorder.
+                InTransform3dChange guard(cmd.path());
+                auto usdSceneItem = std::dynamic_pointer_cast<UsdSceneItem>(cmd.sceneItem());
+                TF_AXIOM(usdSceneItem);
+                UsdGeomXformable xformable(usdSceneItem->prim());
+
+                auto r = xformable.AddRotateXYZOp(UsdGeomXformOp::PrecisionFloat, opSuffix);
+                TF_AXIOM(r);
+                r.Set(v);
+                auto result = setXformOpOrderFn(xformable);
+                TF_AXIOM(result);
+
+                return r;
+            }
+        });
```

Comment:
Combined two lambda OpFunc into one. First check if the prim already has the attribute. If yes, then use it to create UsdGeomXformOp. Otherwise first add the attribute and then use the newly added one. Same changes made in all the command methods.